src/mame/drivers/meritum.cpp

```diff
@@ -215,8 +247,35 @@ uint32_t meritum_state::screen_update_meritum(screen_device &screen, bitmap_ind1
 			{
 				chr = m_p_videoram[x];
 
-				/* get pattern of pixels for that character scanline */
-				gfx = m_p_chargen[(chr<<4) | ra];
+				if( 1 == m_model )
+				{
+					if( chr & 0x80 ) // semigraphics
+						chr = (chr & 0x3F) | 0x40;
+					else if( 0x40 == (chr & 0x60) ) // upper case
+						chr = (chr & 0x3F);
+					else if( 0x60 == (chr & 0x60) ) // lower case
+						chr = (chr & 0x1F ); //use upper
+
+					// shift down comma and semicolon
+					// not sure Meritum I got the circuit for this (like TRS80)
+					// but position of ';' suggests most likely yes
+					if ((chr==',')||(chr==';'))
+					{
+						if ((ra < 10) && (ra > 1))
+							gfx = m_p_chargen[(chr<<4) | (ra-2) ];
+						else
+							gfx = 0;
+					}
+					else
+					{
+						gfx = m_p_chargen[ (chr << 4) | ra ];
+					}
+				}
+				else // Model 2
+				{
+					/* get pattern of pixels for that character scanline */
+					gfx = m_p_chargen[(chr<<4) | ra];
+				}
```

Comment:
Please use a virtual method or something to avoid needing to use a machine type member.